### PR TITLE
Improvement: IPv4, IPv6 and host name can be changed after calling Find Kodi once

### PR DESCRIPTION
**Pull request by wutschel**
## Description

Suggested by @kambala-decapitator.

"Find Kodi" now only needs to be called once. This is will execute the service discovery for _all_ supported address types. Afterwards the user can select the address type without running "Find Kodi" again.

Important: MAC address can only be resolved using iPv4. The code has been changed to use only IPv4 for this, even if the user selects IPv6 or host name as address type. If IPv4 address cannot be found, the MAC address cannot be resolved.

## Summary for release notes

Improvement: IPv4, IPv6 and host name can be selected after calling Find Kodi once

### CODE REVIEW

**Comment by wutschel** [comment]
@kambala-decapitator, I am making this draft as it would be good, if you can test this. I don't want to mess this up right before the release.

File: XBMC Remote/HostViewController.m [hunk 1]
```diff
@@ -88,7 +88,7 @@ - (IBAction)dismissView:(id)sender {
     
     [self textFieldDoneEditing:nil];
     
-    for (UILabel *label in [self getAllEntryMaskLabels]) {
+    for (UITextField *label in [self getAllEntryMaskLabels]) {
```
Comment (kambala-decapitator) [inline XBMC Remote/HostViewController.m:91]: it's not a `label` anymore (in all 3 places)

**Review by kambala-decapitator — APPROVED** [review]
seems to work fine!

you may also want to fix typos in the first commit message

File: XBMC Remote/HostViewController.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/HostViewController.m]: Done

**Comment by wutschel** [comment]
> you may also want to fix typos in the first commit message

Thanks, fixed this as well.

**Comment by wutschel** [comment]
Reworked, squashed and rebased to master.

**Pull request merged**